include/click/atomic.hh

```diff
@@ -76,15 +95,16 @@ class atomic_uint32_t { public:
     inline uint32_t fetch_and_add(uint32_t delta);
     inline bool dec_and_test();
     inline bool nonatomic_dec_and_test();
-    inline uint32_t compare_swap(uint32_t expected, uint32_t desired);
-    inline bool compare_and_swap(uint32_t expected, uint32_t desired) CLICK_DEPRECATED;
+    inline uint32_t compare_swap(uint32_t expected, uint32_t desired) CLICK_BUILTINS_DEPRECATED;
+    inline bool compare_and_swap(uint32_t expected, uint32_t desired) CLICK_BUILTINS_NONDEPRECATED;
 
     inline static uint32_t swap(volatile uint32_t &x, uint32_t desired);
     inline static void inc(volatile uint32_t &x);
     inline static bool dec_and_test(volatile uint32_t &x);
-    inline static uint32_t compare_swap(volatile uint32_t &x, uint32_t expected, uint32_t desired);
-    inline static bool compare_and_swap(volatile uint32_t &x, uint32_t expected, uint32_t desired) CLICK_DEPRECATED;
+    inline static uint32_t compare_swap(volatile uint32_t &x, uint32_t expected, uint32_t desired) CLICK_BUILTINS_DEPRECATED;
```

Comment:
What is the one to use now then? :p

Reply:
The previous Click code used `uint32_t compare_swap`: https://github.com/tbarbette/fastclick/commit/4fdcf82c659fbff108906b6a97d48972e9b9291a.
However, it is fully atomic only on x86 implementation.

The rationale behind the flags and the various `#if CLICK_COMPARE_SWAP` around the code is to maintain compatibility:
- if `--enable-atomic-builtins` is given, `compare_swap` is marked as deleted and all the calls are replaced with the equivalent `compare_and_swap`
- by default, the old deprecated mark on `compare_and_swap` is maintained, while `compare_swap` is used in the calls.

I think that there isn't any usage in the code that requires `compare_swap` instead of `compare_and_swap` and all the occurrences could be replaced by `compare_and_swap`.